src/coreclr/src/vm/jithelpers.cpp

```diff
@@ -4783,6 +4783,10 @@ HCIMPLEND
 /*************************************************************/
 HCIMPL0(VOID, JIT_PollGC)
 {
+    // as long as we can have GCPOLL_CALL polls, it would not hurt to check the trap flag.
```

Comment:
We can also split the method into a fast and slow NOINLINE parts to avoid frame setup cost for the fast path, similar to how other JIT helpers are done.